### DISCUSSION TITLE
Examples segfault when run

### DISCUSSION THREAD

**Discussion by 8bitprodigy** [CuarzoSoftware/Louvre#28] [Support] [answered]
I installed Louvre and SRM following the instructions here:  https://cuarzosoftware.github.io/Louvre/md_md__downloads.html

But for whatever reason, when I try to run any of them, I get a segfault.
Weston does work, however.
I'm running Kubuntu 23.10, not sure what other info to provide, because it doesn't provide any other error messages, and the screen flicks black for like a frame or two.

**Comment by ehopperdietzel** [comment, marked as answer]
Hello, it appears that you're not executing them from a free TTY. Try using the command `chvt N` or pressing `Ctrl + Alt + F[1,2,..,N]` to switch to another virtual terminal. If you're unable to do so, consider restarting. In some instances, launching it from an X11 desktop like Cinnamon may interfere with the seatd service, causing issues with session switching.

Additionally, enabling debugging messages can be achieved by setting the **SRM_DEBUG** and **LOUVRE_DEBUG** environment variables to 4.

**Reply by 8bitprodigy** [reply]
> Ctrl + Alt + F[1,2,..,N]

Okay, so I have to switch to the cli. I get it now. Thanks!

Would be great for development if I can launch it in a window for testing like weston does, but this isn't that hard either.

**Reply by ehopperdietzel** [reply]
Awesome! Indeed, I'm aware that this complicates development and debugging. Previously, there was an X11 backend, but I haven't yet migrated it to the new backend API. My main focus is on the DRM backend, which is crucial. Once it's polished and generic enough, my plan is to port the X11 backend and incorporate a Wayland backend as well.

**Reply by 8bitprodigy** [reply]
Fantastic! Looking forward to it!
